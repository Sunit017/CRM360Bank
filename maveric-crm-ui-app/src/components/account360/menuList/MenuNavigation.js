import React, { useState } from "react";
import { Button, Menu, MenuItem } from "@mui/material";
import styled from "styled-components";
import { KeyboardArrowDown } from "@mui/icons-material";
import { isMobile } from "react-device-detect";

const Header = styled.div`
  // padding: 8px;
  background: #999;
  // border-radius: 10px;
  // width: calc(100vw - 90px);
  // margin-bottom: 10px;
  display: flex;
  flex-direction: row;
`;

const MenuNavigation = ({ selectedMenu }) => {
  const [eleE1, setEleE1] = useState(null);
  const [eleE2, setEleE2] = useState(null);

  const handleAccount360Close = (item = {}) => {
    setEleE1(null);
    if (item?.id) {
      selectedMenu(item);
    }
  };
  const onshowAccount360Menu = (evt) => {
    setEleE1(evt.currentTarget);
  };
  const onshowMakerCheckerMenu = (evt) => {
    setEleE2(evt.currentTarget);
    // selectedMenu({ id: "makeDepositWithdrawal" });
  };
  const handleMakerCheckerClose = (item = null) => {
    setEleE2(null);
    if (item) {
      selectedMenu(item);
    }
  };

  const menuList = [
    {
      id: "makerChecker",
      menuLabel: "Maker Checker",
      childItem: [
        {
          id: "makeDepositWithdrawal",
          label: "Make Deposit Withdrawal",
        },
        {
          id: "createAppoveClients",
          label: "Create Appove Clients",
        },
        {
          id: "approveCloseLockUnlockAccount",
          label: "Approve / Close / Lock / Unlock Account",
        },
        {
          id: "applyAccountFees",
          label: "Apply Account Fees",
        },
        {
          id: "applyAccuredInterest",
          label: "Apply Accured Interest",
        },
        {
          id: "customer360",
          label: "Customer 360",
        },
      ],
    },
  ];

  return (
    <Header>
      <Button
        endIcon={<KeyboardArrowDown />}
        variant="containedMenu"
        style={{ color: "#fff", textTransform: "none" }}
        onClick={(evt) => onshowAccount360Menu(evt)}
        data-testid="Account360Dropdown"
      >
        Account 360
      </Button>
      <Button
        endIcon={<KeyboardArrowDown />}
        variant="containedMenu"
        style={{ color: "#fff", textTransform: "none" }}
        onClick={(evt) => onshowMakerCheckerMenu(evt)}
        data-testid="MakerCheckerDropdown"
      >
        Maker Checker
      </Button>

      <Menu
        data-testid="account360-menu"
        open={Boolean(eleE1)}
        anchorEl={eleE1}
        onClose={() => handleAccount360Close()}
        PaperProps={{
          style: {
            // maxHeight: 40 * 4,
            width: "20ch",
          },
        }}
      >
        <MenuItem
          sx={{
            fontSize: "small",
            minHeight: "32px",
          }}
          onClick={() => {
            handleAccount360Close({ id: "accountDetails" });
          }}
          data-testid="AccountDetailsMenuItem"
        >
          Account Details
        </MenuItem>
        <MenuItem
          sx={{
            fontSize: "small",
            minHeight: "32px",
          }}
          onClick={() => {
            handleAccount360Close({ id: "accountCardDetails" });
          }}
          data-testid="CardDetailsMenuItem"
        >
          Card Details
        </MenuItem>
      </Menu>
      <Menu
        data-testid="makerChecker-menu"
        open={Boolean(eleE2)}
        anchorEl={eleE2}
        onClose={() => handleMakerCheckerClose()}
        PaperProps={{
          style: {
            height: "110px",
            width: "calc(100vw - 100px)",
          },
        }}
      >
        {menuList[0].childItem.map((item) => {
          const menuCss = !isMobile
            ? {
                marginRight: "0px",
                position: "relative",
                minHeight: "32px",
                width: "50%",
                float: "left",
              }
            : {
                minHeight: "35px",
              };
          return (
            <MenuItem
              sx={{
                fontSize: "small",
                ...menuCss,
              }}
              onClick={() => {
                handleMakerCheckerClose(item);
              }}
              data-testid="MakerCheckerMenuItems"
              key={item.id}
            >
              {item.label}
            </MenuItem>
          );
        })}
      </Menu>
    </Header>
  );
};

export default MenuNavigation;
