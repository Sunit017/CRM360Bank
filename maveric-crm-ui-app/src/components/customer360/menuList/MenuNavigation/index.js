import React, { useState } from "react";
import { Button, Menu, MenuItem, Grid } from "@mui/material";
import styled from "styled-components";
import { removeAccout360Page } from "../../../../store/actions";
import { KeyboardArrowDown } from "@mui/icons-material";
import { useDispatch } from "react-redux";
import MakerCheckerMenu from "../../../makerChecker/opsMenu";
import { useSelector } from "react-redux";
import { isMobile } from "react-device-detect";

const Header = styled.div`
  background: #214E8F;
  margin-bottom: 10px;
`;

const MenuNavigation = ({ selectedMenu }) => {
  const [eleE1, setEleE1] = useState(null);
  const [eleE2, setEleE2] = useState(null);
  const [eleE3, setEleE3] = useState(null);
  const [eleE4, setEleE4] = useState(null);
  const [enbleMakerChecker, setEnbleMakerChecker] = useState(false);

  //To enable maker checker menu
  const accountSelection = useSelector((state) => state.accountSelection);
  const { pageId } = accountSelection || {};

  const dispatch = useDispatch();
  const onshowCustomer360Menu = (evt) => {
    setEleE1(evt.currentTarget);
  };
  const handleCustomer360Close = (item = null) => {
    setEleE1(null);
    if (item) {
      selectedMenu(item);
    }
    setEnbleMakerChecker(false);
  };
  const onshowAccount360Menu = (evt) => {
    dispatch(removeAccout360Page());
    selectedMenu({ id: "account360" });
    setEnbleMakerChecker(true);
  };

  const onshowCase360Menu = (evt) => {
    setEleE2(evt.currentTarget);
  };
  const handleCase360Close = (item = null) => {
    setEleE2(null);
    if (item) {
      selectedMenu(item);
    }
    setEnbleMakerChecker(false);
  };
  const onshowCallContactMenu = (evt) => {
    setEleE3(evt.currentTarget);
  };
  const handleCallContactClose = () => {
    setEleE3(null);
    setEnbleMakerChecker(false);
  };
  const onshowChatContactMenu = (evt) => {
    setEleE4(evt.currentTarget);
  };
  const handleChatContactClose = () => {
    setEleE4(null);
    setEnbleMakerChecker(false);
  };

  const onshowTasksMenu = (evt) => {
    selectedMenu({ id: "tasks" });
    setEnbleMakerChecker(false);
  };

  const makerCheckerMenu = (item) => {
    selectedMenu(item);
  };
  const menuList = [
    {
      id: "customer360",
      menuLabel: "Customer 360",
      childItem: [
        {
          id: "customerDetails",
          label: "Customer Details",
        },
        {
          id: "personalDetails",
          label: "Personal Details",
        },
        {
          id: "permanentAddress",
          label: "Permanent Address",
        },
        {
          id: "businessDetails",
          label: "Business Details",
        },
        {
          id: "businessAddress",
          label: "Business Address",
        },
        {
          id: "employmentDetails",
          label: "Employment Details",
        },
        {
          id: "employmentAddress",
          label: "Employment Address",
        },
        {
          id: "spouseDetails",
          label: "Spouse Details",
        },
        {
          id: "identityDetails",
          label: "Identification Details",
        },
        {
          id: "financialDetails",
          label: "Financial Details",
        },
        {
          id: "nomineeDetails",
          label: "Nominee Details",
        },
        {
          id: "clientStatus",
          label: "Client Status",
        },
        { id: "idMission", label: "IdMission" },
        {
          id: "idMissionDedupe",
          label: "IdMission Dedupe",
        },
        {
          id: "AMLStatus",
          label: "AML Status",
        },
        {
          id: "KYCHighRiskfields",
          label: "KYC High-Risk fields",
        },
        {
          id: "loanFields",
          label: "Loan Fields",
        },
      ],
    },
  ];
  return (
    <>
      <Header
        style={{
          display: "flex",
          flexDirection: isMobile ? "column" : "row",
          alignItems: "start",
        }}
      >
        <Button
          sx={{
            padding: "6px 10px",
          }}
          endIcon={<KeyboardArrowDown />}
          variant="containedMenu"
          style={{ color: "#fff", textTransform: "none" }}
          onClick={(evt) => onshowCustomer360Menu(evt)}
          data-testid="Customer360Dropdown"
          // onMouseEnter={(evt) => onshowCustomer360Menu(evt)}
        >
          Customer 360
        </Button>

        <Button
          sx={{
            padding: "6px 10px",
          }}
          variant="containedMenu"
          style={{ color: "#fff", textTransform: "none" }}
          data-testid="Account360Dropdown"
          onClick={(evt) => onshowAccount360Menu(evt)}
        >
          Account 360
        </Button>
        <Button
          sx={{
            padding: "6px 10px",
          }}
          endIcon={<KeyboardArrowDown />}
          variant="containedMenu"
          style={{ color: "#fff", textTransform: "none" }}
          onClick={(evt) => onshowCase360Menu(evt)}
          data-testid="Case360Dropdown"
          // onMouseEnter={(evt) => onshowCase360Menu(evt)}
        >
          Case 360
        </Button>
        <Button
          sx={{
            padding: "6px 10px",
          }}
          endIcon={<KeyboardArrowDown />}
          variant="containedMenu"
          style={{ color: "#fff", textTransform: "none" }}
          onClick={(evt) => onshowCallContactMenu(evt)}
          data-testid="CallContactHistoryDropdown"
          // onMouseEnter={(evt) => onshowCallContactMenu(evt)}
        >
          Call Contact History
        </Button>
        <Button
          sx={{
            padding: "6px 10px",
          }}
          endIcon={<KeyboardArrowDown />}
          variant="containedMenu"
          style={{ color: "#fff", textTransform: "none" }}
          onClick={(evt) => onshowChatContactMenu(evt)}
          data-testid="ChatContactHistoryDropdown"
          // onMouseEnter={(evt) => onshowChatContactMenu(evt)}
        >
          Chat Contact History
        </Button>
        <Button
          sx={{
            padding: "6px 10px",
          }}
          //endIcon={<KeyboardArrowDown />}
          variant="containedMenu"
          style={{ color: "#fff", textTransform: "none" }}
          onClick={(evt) => onshowTasksMenu(evt)}
          data-testid="TasksDropdown"
          // onMouseEnter={(evt) => onshowChatContactMenu(evt)}
        >
          Tasks
        </Button>
        <Menu
          data-testid="customer360-menu"
          open={Boolean(eleE1)}
          anchorEl={eleE1}
          onClose={() => handleCustomer360Close()}
          PaperProps={{
            style: {
              height: "210px",
              width: "calc(100vw - 80px)",
            },
          }}
        >
          {menuList[0].childItem.map((item) => {
            const menuCss = !isMobile
              ? {
                  marginRight: "30px",
                  position: "relative",
                  minHeight: "32px",
                  width: "25%",
                  float: "left",
                }
              : {
                  minHeight: "35px",
                };
            return (
              <MenuItem
                style={{
                  fontSize: "small",
                  ...menuCss,
                }}
                onClick={() => {
                  // navigate("/app/case360/management");
                  handleCustomer360Close(item);
                }}
                data-testid="Customer360MenuItems"
                key={item.id}
              >
                {item.label}
              </MenuItem>
            );
          })}
        </Menu>

        <Menu
          data-testid="cae360-menu"
          open={Boolean(eleE2)}
          anchorEl={eleE2}
          onClose={() => handleCase360Close()}
          PaperProps={{
            style: {
              maxHeight: 40 * 4,
              width: "20ch",
            },
          }}
        >
          <MenuItem
            sx={{
              fontSize: "small",
              minHeight: "32px",
            }}
            onClick={() => {
              // navigate("/app/case360/management");
              handleCase360Close({ id: "cases" });
            }}
            data-testid="CasesMenuItem"
          >
            Cases
          </MenuItem>
          <MenuItem
            sx={{
              fontSize: "small",
              minHeight: "32px",
            }}
            onClick={() => {
              // navigate("/app/case360/management");
              handleCase360Close({ id: "createcase" });
            }}
          >
            Create Case
          </MenuItem>
        </Menu>
        <Menu
          data-testid="callContact-menu"
          open={Boolean(eleE3)}
          anchorEl={eleE3}
          onClose={() => handleCallContactClose()}
          PaperProps={{
            style: {
              maxHeight: 40 * 4,
              width: "20ch",
            },
          }}
        >
          <MenuItem
            sx={{
              fontSize: "small",
              minHeight: "32px",
            }}
            onClick={() => {
              // navigate("/app/case360/management");
              handleCallContactClose();
            }}
            data-testid="CallHistoryMenuItem"
          >
            Call History
          </MenuItem>
        </Menu>
        <Menu
          data-testid="chatContact-menu"
          open={Boolean(eleE4)}
          anchorEl={eleE4}
          onClose={() => handleChatContactClose()}
          PaperProps={{
            style: {
              maxHeight: 40 * 4,
              width: "20ch",
            },
          }}
        >
          <MenuItem
            sx={{
              fontSize: "small",
              minHeight: "32px",
            }}
            onClick={() => {
              // navigate("/app/case360/management");
              handleChatContactClose();
            }}
            data-testid="ChatHistoryMenuItem"
          >
            Chat History
          </MenuItem>
        </Menu>
      </Header>

      {/* maker checker menu enabled on click of account no. */}
      {enbleMakerChecker && pageId === "accountDetails" && (
        <MakerCheckerMenu onSelectMenu={selectedMenu} />
      )}
    </>
  );
};

export default MenuNavigation;
