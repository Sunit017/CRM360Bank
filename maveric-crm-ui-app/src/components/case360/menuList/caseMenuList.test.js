import renderWithRedux from "../../../testUtil";
import { cleanup, fireEvent, screen } from "@testing-library/react";
import MenuNavigation from "../menuList/index";

afterEach(cleanup);

test("Render Menu List container", () => {
  const container = renderWithRedux(<MenuNavigation />, {});
  expect(container).toMatchSnapshot();
});

test("Click Case 360 Menu Navigation view", () => {
    renderWithRedux(<MenuNavigation />, {});
    fireEvent.click(screen.getByTestId("btn-showcase"));
  });

test("Click Case 360 Menu Item view", () => {
    renderWithRedux(<MenuNavigation />, {});
    fireEvent.click(screen.getByTestId("btn-showcase"));
    fireEvent.click(screen.getByTestId("case360-menu"));
    // fireEvent.click(screen.getByTestId("case360-menu1"));
  });

  
