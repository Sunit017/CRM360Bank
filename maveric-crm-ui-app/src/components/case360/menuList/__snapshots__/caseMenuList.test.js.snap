// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Render Menu List container 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="sc-bczRLJ huZbBx"
      >
        <button
          class="MuiButtonBase-root MuiButton-root MuiButton-containedMenu MuiButton-containedMenuPrimary MuiButton-sizeMedium MuiButton-containedMenuSizeMedium css-1w1rijm-MuiButtonBase-root-MuiButton-root"
          data-testid="btn-showcase"
          style="color: rgb(255, 255, 255); text-transform: none;"
          tabindex="0"
          type="button"
        >
          Case 360
          <span
            class="MuiButton-endIcon MuiButton-iconSizeMedium css-9tj150-MuiButton-endIcon"
          >
            <svg
              aria-hidden="true"
              class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
              data-testid="KeyboardArrowDownIcon"
              focusable="false"
              viewBox="0 0 24 24"
            >
              <path
                d="M7.41 8.59 12 13.17l4.59-4.58L18 10l-6 6-6-6 1.41-1.41z"
              />
            </svg>
          </span>
          <span
            class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
          />
        </button>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="sc-bczRLJ huZbBx"
    >
      <button
        class="MuiButtonBase-root MuiButton-root MuiButton-containedMenu MuiButton-containedMenuPrimary MuiButton-sizeMedium MuiButton-containedMenuSizeMedium css-1w1rijm-MuiButtonBase-root-MuiButton-root"
        data-testid="btn-showcase"
        style="color: rgb(255, 255, 255); text-transform: none;"
        tabindex="0"
        type="button"
      >
        Case 360
        <span
          class="MuiButton-endIcon MuiButton-iconSizeMedium css-9tj150-MuiButton-endIcon"
        >
          <svg
            aria-hidden="true"
            class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
            data-testid="KeyboardArrowDownIcon"
            focusable="false"
            viewBox="0 0 24 24"
          >
            <path
              d="M7.41 8.59 12 13.17l4.59-4.58L18 10l-6 6-6-6 1.41-1.41z"
            />
          </svg>
        </span>
        <span
          class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
        />
      </button>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "store": Object {
    "@@observable": [Function],
    "dispatch": [Function],
    "getState": [Function],
    "replaceReducer": [Function],
    "subscribe": [Function],
  },
  "unmount": [Function],
}
`;
